### PR TITLE
Create orders

### PR DESCRIPTION
**Pull request by kiran-mastek**
Bug: Multiple selections of proxy orgs causes error

Add caching to 3rd party API call to prevent error in dev environment

Story: [AB#12959](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/12959)
Task: [AB#14971](https://buyingcatalog.visualstudio.com/c5f97979-5b03-4d10-ba8d-871d0526b408/_workitems/edit/14971)

### CODE REVIEW

File: src/NHSD.BuyingCatalogue.Organisations.Api/Repositories/OdsRepository.cs [hunk 1]
```diff
@@ -14,19 +15,23 @@ namespace NHSD.BuyingCatalogue.Organisations.Api.Repositories
...
 
         public async Task<OdsOrganisation> GetBuyerOrganisationByOdsCodeAsync(string odsCode)
         {
-            var response = await settings.ApiBaseUrl
-                .AppendPathSegment("organisations")
-                .AppendPathSegment(odsCode)
-                .AllowHttpStatus("3xx,4xx")
-                .GetJsonAsync<OdsResponse>();
+            var response = await appCache.GetOrAddAsync(
+                $"Ods-Org-{odsCode}",
```
Comment (bjss-chris) [inline src/NHSD.BuyingCatalogue.Organisations.Api/Repositories/OdsRepository.cs:29]: Is there a reason why we can't just use the ODS code as the key given that we're not currently using any other type of key?

File: tests/NHSD.BuyingCatalogue.Organisations.Api.IntegrationTests/Features/Ods Organisations GetByOdsCode.feature [hunk 2]
```diff
@@ -77,6 +77,7 @@ Scenario: If a user is not authorised then they cannot access the organisation
     When a GET request is made for an Ods organisation with code B1G
     Then a response with status code 401 is returned
 
+@ignore
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Organisations.Api.IntegrationTests/Features/Ods Organisations GetByOdsCode.feature:80]: Can you add a brief comment explaining why this scenario is ignored?

File: tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/Repository/OdsRepositoryTests.cs [hunk 3]
```diff
@@ -5,8 +5,10 @@
 using FluentAssertions;
 using Flurl.Http;
 using Flurl.Http.Testing;
+using LazyCache;
 using NHSD.BuyingCatalogue.Organisations.Api.Models;
 using NHSD.BuyingCatalogue.Organisations.Api.Repositories;
+using NHSD.BuyingCatalogue.Organisations.Api.Settings;
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/Repository/OdsRepositoryTests.cs:11]: Do we need `Settings`?

File: tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/Repository/OdsRepositoryTests.cs [hunk 4]
```diff
@@ -115,7 +133,7 @@ public static async Task GetBuyerOrganisationByOdsCode_CallsOdsApi_Once()
         [SuppressMessage("ReSharper", "ObjectCreationAsStatement", Justification = "Testing")]
         public static void Constructor_IOdsRepository_OdsSettings_NullSettings_ThrowsException()
         {
-            Assert.Throws<ArgumentNullException>(() => new OdsRepository(null));
+            Assert.Throws<ArgumentNullException>(() => new OdsRepository(null, new LazyCache.CachingService()));
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/Repository/OdsRepositoryTests.cs:136]: Is `LazyCache.` redundant here? Can we simplify this to `new CachingService()`?

File: tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/TestContexts/OdsRepositoryTestContext.cs [hunk 5]
```diff
@@ -13,7 +13,9 @@ private OdsRepositoryTestContext()
                 BuyerOrganisationRoleIds = new[] { "RO98", "RO177", "RO213", "RO272" },
             };
 
-            OdsRepository = new OdsRepository(OdsSettings);
+            LazyCache.CachingService.DefaultCacheProvider.Value.Remove($"Ods-Org-XYZ");
```
Comment (bjss-chris) [inline tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/TestContexts/OdsRepositoryTestContext.cs:16]: Looks like a redundant `$` here as we're not doing any interpolation.

**Review by bjss-chris — CHANGES_REQUESTED** [review]

File: tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/Repository/OdsRepositoryTests.cs (hunk 3 above)
Comment (kiran-mastek) [reply tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/Repository/OdsRepositoryTests.cs:11]: Removed

File: src/NHSD.BuyingCatalogue.Organisations.Api/Repositories/OdsRepository.cs (hunk 1 above)
Comment (kiran-mastek) [reply src/NHSD.BuyingCatalogue.Organisations.Api/Repositories/OdsRepository.cs:29]: changed

File: tests/NHSD.BuyingCatalogue.Organisations.Api.IntegrationTests/Features/Ods Organisations GetByOdsCode.feature (hunk 2 above)
Comment (kiran-mastek) [reply tests/NHSD.BuyingCatalogue.Organisations.Api.IntegrationTests/Features/Ods Organisations GetByOdsCode.feature:80]: done

File: tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/Repository/OdsRepositoryTests.cs (hunk 4 above)
Comment (kiran-mastek) [reply tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/Repository/OdsRepositoryTests.cs:136]: done

File: tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/TestContexts/OdsRepositoryTestContext.cs (hunk 5 above)
Comment (kiran-mastek) [reply tests/NHSD.BuyingCatalogue.Organisations.Api.UnitTests/TestContexts/OdsRepositoryTestContext.cs:16]: removed

**Review by bjss-chris — APPROVED** [review]

**Pull request merged**